massa-grpc/src/server.rs

```diff
@@ -75,22 +77,49 @@ impl MassaGrpc {
             .max_frame_size(config.max_frame_size);
 
         if config.accept_http1 {
-            let mut router_with_http1 = server_builder
-                .accept_http1(true)
-                .layer(GrpcWebLayer::new())
-                .add_service(svc);
-
-            if config.enable_reflection {
-                let reflection_service = tonic_reflection::server::Builder::configure()
-                    .register_encoded_file_descriptor_set(FILE_DESCRIPTOR_SET)
-                    .build()?;
-
-                router_with_http1 = router_with_http1.add_service(reflection_service);
+            if config.enable_cors {
+                let cors = CorsLayer::new()
+                    // Allow `GET`, `POST` and `OPTIONS` when accessing the resource
+                    .allow_methods([Method::GET, Method::POST, Method::OPTIONS])
+                    // Allow requests from any origin
+                    .allow_origin(Any)
+                    .allow_headers([hyper::header::CONTENT_TYPE]);
+
+                let mut router_with_http1 = server_builder
+                    .accept_http1(true)
+                    .layer(cors)
+                    .layer(GrpcWebLayer::new())
+                    .add_service(svc);
+
+                if config.enable_reflection {
```

Comment:
maybe can refactor the build of router_with_http1 for avoid duplication

Reply:
I have the same issue.. builder is not a builder pattern but it's a typestate pattern. It's impossible to optimize